azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.resource;
+
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.microsoft.azure.management.resources.fluentcore.arm.ResourceId;
+import com.microsoft.azure.toolkit.lib.Azure;
+import com.microsoft.azure.toolkit.lib.AzureConfiguration;
+import com.microsoft.azure.toolkit.lib.AzureService;
+import com.microsoft.azure.toolkit.lib.SubscriptionScoped;
+import com.microsoft.azure.toolkit.lib.auth.Account;
+import com.microsoft.azure.toolkit.lib.auth.AzureAccount;
+import com.microsoft.azure.toolkit.lib.common.cache.Cacheable;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.model.Subscription;
+import com.microsoft.azure.toolkit.lib.common.task.ICommittable;
+import org.apache.commons.lang3.StringUtils;
+import reactor.core.publisher.Flux;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class AzureResourceGroup extends SubscriptionScoped<AzureResourceGroup> implements AzureService {
+
+    public AzureResourceGroup() { // for SPI
+        super(AzureResourceGroup::new);
+    }
+
+    private AzureResourceGroup(@Nonnull final List<Subscription> subscriptions) {
+        super(AzureResourceGroup::new, subscriptions);
+    }
+
+    public List<ResourceGroupEntity> list() {
+        return Flux.fromIterable(getSubscriptions()).parallel()
+                .map(subscription -> getAzureResourceManager(subscription.getId()))
+                .flatMap(azureResourceManager -> azureResourceManager.resourceGroups().listAsync())
+                .map(ResourceGroupEntity::fromResource)
+                .sequential().collectList().block();
+    }
+
+    public ResourceGroupEntity getById(@Nonnull String idStr) {
+        final ResourceId id = ResourceId.fromString(idStr);
+        return get(id.subscriptionId(), id.resourceGroupName());
+    }
+
+    public ResourceGroupEntity getByName(@Nonnull String name) {
+        return get(getDefaultSubscription().getId(), name);
+    }
+
+    public ResourceGroupEntity get(@Nonnull String subscriptionId, @Nonnull String name) {
+        return ResourceGroupEntity.fromResource(getAzureResourceManager(subscriptionId).resourceGroups().getByName(name));
```

Comment:
As sdk will throw exception for none exist resources, do we need to handle this exception and return null here? Or do we need to add an `exist()` method to determine whether resource group with specific name exists?

Reply:
throw exception will be fine, null for non-existing is to abort exception handling, we don't have the need to check exist?

Reply:
will add exist later